[Calc] Remove container-progress() and media-progress() functions

weinig opened the pull request:
#### f3826ea216442e73706ae3a320992c3f9b6e02e8
<pre>
[Calc] Remove container-progress() and media-progress() functions
<a href="https://bugs.webkit.org/show_bug.cgi?id=294736">https://bugs.webkit.org/show_bug.cgi?id=294736</a>

Reviewed by Anne van Kesteren.

Removes support for the container-progress() and media-progress() functions
which have both been removed from the CSS Values 5 spec.

Both of these have been off by default and have never shipped enabled.

* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-computed.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-computed.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-invalid.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-invalid.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-serialize.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-serialize.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-computed.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-computed.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-invalid.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-invalid.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-serialize.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-serialize.tentative.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ComputedStyleDependencies.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.cpp: Removed.
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.h: Removed.
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Evaluation.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Mappings.h:
* Source/WebCore/css/calc/CSSCalcTree+MediaProgressEvaluator.cpp: Removed.
* Source/WebCore/css/calc/CSSCalcTree+MediaProgressEvaluator.h: Removed.
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.h:
* Source/WebCore/css/calc/CSSCalcTree+Traversal.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSParserContext.h:
* Source/WebCore/css/query/ContainerQueryFeatures.cpp:
* Source/WebCore/css/query/ContainerQueryFeatures.h:
* Source/WebCore/css/query/ContainerQueryParser.cpp:
* Source/WebCore/css/query/ContainerQueryParser.h:
* Source/WebCore/css/query/MediaQueryFeatures.cpp:
* Source/WebCore/css/query/MediaQueryFeatures.h:
* Source/WebCore/css/query/MediaQueryParser.cpp:
* Source/WebCore/css/query/MediaQueryParser.h:

Canonical link: <a href="https://commits.webkit.org/296426@main">https://commits.webkit.org/296426@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/de8a7d8883bb7378873bc5b4ebe60721a9bff80b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28164 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18588 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113712 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58902 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28853 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36716 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82403 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111451 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22892 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97732 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62839 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22309 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15870 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58428 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/101060 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/92262 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15923 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116833 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/107062 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35557 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/26203 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91427 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35930 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94006 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91228 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23242 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36124 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13887 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31304 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35459 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40994 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/131344 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35168 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35637 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38520 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36849 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
Assuming I understood correctly these are prototype implementations that never shipped enabled-by-default. (It would be good to add that to the commit message to avoid further confusion.)

weinig committed:
[Calc] Remove container-progress() and media-progress() functions
https://bugs.webkit.org/show_bug.cgi?id=294736

Reviewed by Anne van Kesteren.

Removes support for the container-progress() and media-progress() functions
which have both been removed from the CSS Values 5 spec.

Both of these have been off by default and have never shipped enabled.

* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-computed.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-computed.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-invalid.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-invalid.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-serialize.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/container-progress-serialize.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-computed.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-computed.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-invalid.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-invalid.tentative.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-serialize.tentative-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-values/media-progress-serialize.tentative.html: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/calc/CSSCalcTree+CalculationValue.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ComputedStyleDependencies.cpp:
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.cpp: Removed.
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.h: Removed.
* Source/WebCore/css/calc/CSSCalcTree+Copy.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Evaluation.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Mappings.h:
* Source/WebCore/css/calc/CSSCalcTree+MediaProgressEvaluator.cpp: Removed.
* Source/WebCore/css/calc/CSSCalcTree+MediaProgressEvaluator.h: Removed.
* Source/WebCore/css/calc/CSSCalcTree+Parser.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Serialization.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.cpp:
* Source/WebCore/css/calc/CSSCalcTree+Simplification.h:
* Source/WebCore/css/calc/CSSCalcTree+Traversal.h:
* Source/WebCore/css/calc/CSSCalcTree.cpp:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/parser/CSSParserContext.cpp:
* Source/WebCore/css/parser/CSSParserContext.h:
* Source/WebCore/css/query/ContainerQueryFeatures.cpp:
* Source/WebCore/css/query/ContainerQueryFeatures.h:
* Source/WebCore/css/query/ContainerQueryParser.cpp:
* Source/WebCore/css/query/ContainerQueryParser.h:
* Source/WebCore/css/query/MediaQueryFeatures.cpp:
* Source/WebCore/css/query/MediaQueryFeatures.h:
* Source/WebCore/css/query/MediaQueryParser.cpp:
* Source/WebCore/css/query/MediaQueryParser.h:

Canonical link: https://commits.webkit.org/296426@main
